clients/src/main/java/org/apache/kafka/common/PartitionInfo.java

```diff
@@ -19,20 +19,28 @@
 /**
  * Information about a topic-partition.
  */
+// TODO: rename this class to MetadataPartitionInfo
```

Comment:
This patch adds the Scala class `LeaderAndIsrPartitionState` and `MetadataPartitionState`. `MetadataPartitionInfo` makes it more explicit that this is a Java class that corresponds to `MetadataPartitionState`.